tests/test_bt_merge.cpp

```diff
@@ -1,12 +1,76 @@
+#include <rapidcheck.h>
+
 #include <catch2/catch_test_macros.hpp>
 
 #include "catch2_bt_format.hpp"
+#include "rapidcheck/catch.h"
 #include "session/bt_merge.hpp"
 
 using oxenc::bt_dict;
 using oxenc::bt_list;
 using oxenc::bt_value;
 
+namespace rc {
+
+template <>
+class Arbitrary<std::string_view> {
+  public:
+    static rc::Gen<std::string_view> arbitrary() {
+
+        return rc::gen::map(rc::gen::arbitrary<std::string>(), [](const std::string& s) {
+            return std::string_view(s);
+        });
+    }
+};
+
+Gen<bt_value> genBtValue() {
+    return gen::oneOf<bt_value>(
+            gen::map(
+                    gen::arbitrary<std::string>(),
+                    [](const std::string& s) { return bt_value{s}; }),
+            gen::map(
+                    gen::arbitrary<std::string_view>(),
```

Comment:
usage of `std::string_view` here is dubious as we have no concept of lifetime of the underlying data here.

Reply:
This one makes me think and I still don't have an answer, I lack the knowledge of some C++ fundamentals.

I think a safe way is like this:
```
template <>
class Arbitrary<std::string_view> {
  public:
    static rc::Gen<std::string_view> arbitrary() {
       return gen::just(std::string_view("string view placeholder"));
    }
};
``` 
which refers to some static string literals so we don't have to worry about the life cycle. Unfortunately, this can't generate random string_view dynamically.

I'm not sure the previous one is wrong either, not sure how RapidCheck manages the life cycle of std::string it creates, but it seems risky to rely on implementation details.

What's your suggestion? Any hints are appreciated.

Reply:
```
rc::check("idempotence", [](const bt_dict& x) { REQUIRE(merge(x, x) == x); });
```

My understanding is, every time when a new `bt_dict& x` is generated, a chain(or DAG) of generators manage their dependencies and the life cycle of the values they created. Any element in this chain should guarantee to be alive during a single test run in the `REQUIRE` block, then everything is freed and a new test run is started. I think our usage of `std::string_view` should be safe. Does that make sense?